### ISSUE TITLE
Custom Fields: Invalid date format crashes workflow

### ISSUE THREAD

**Issue by MarcoWel** [freescout-helpdesk/freescout#2239]
We use the Custom Fields module and use Workflows to check for those fields automatically.

When having a date field for messages, the date can entered in the field as free text in the Freescout backend:
![image](https://user-images.githubusercontent.com/80592577/190232493-bbc31bc4-9417-417c-8c71-6fa1d7368781.png)

If this text is an invalid date string with the current locale, workflows accessing the date field keep crashing until the date field of the particular message is corrected or cleared:

`App\Jobs\TriggerAction has been attempted too many times or run too long. The job may have previously timed out. {"exception":"[object] (Illuminate\\Queue\\MaxAttemptsExceededException(code: 0): App\\Jobs\\TriggerAction has been attempted too many times or run too long. The job may have previously timed out. at /var/www/html/vendor/laravel/framework/src/Illuminate/Queue/Worker.php:394)`

and:

```
DateTime::__construct(): Failed to parse time string (xxxxxxxxxxx) at position 0 (x): The timezone could not be found in the database {"exception":"[object] (Exception(code: 0): DateTime::__construct(): Failed to parse time string (xxxxxxxxxxx) at position 0 (x): The timezone could not be found in the database at /var/www/html/overrides/nesbot/carbon/src/Carbon/Carbon.php:548)
[stacktrace]
#0 /var/www/html/overrides/nesbot/carbon/src/Carbon/Carbon.php(548): DateTime->__construct()
#1 /var/www/html/overrides/nesbot/carbon/src/Carbon/Carbon.php(587): Carbon\\Carbon->__construct()
#2 /var/www/html/Modules/CustomFields/Providers/CustomFieldsServiceProvider.php(324): Carbon\\Carbon::parse()
#3 [internal function]: Modules\\CustomFields\\Providers\\CustomFieldsServiceProvider->Modules\\CustomFields\\Providers\\{closure}()
#4 /var/www/html/overrides/tormjens/eventy/src/Filter.php(28): call_user_func_array()
#5 /var/www/html/overrides/laravel/framework/src/Illuminate/Support/Collection.php(339): TorMorten\\Eventy\\Filter->TorMorten\\Eventy\\{closure}()
#6 /var/www/html/overrides/tormjens/eventy/src/Filter.php(29): Illuminate\\Support\\Collection->each()
#7 /var/www/html/vendor/tormjens/eventy/src/Events.php(163): TorMorten\\Eventy\\Filter->fire()
#8 /var/www/html/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(221): TorMorten\\Eventy\\Events->filter()
#9 /var/www/html/Modules/Workflows/Entities/Workflow.php(984): Illuminate\\Support\\Facades\\Facade::__callStatic()
#10 /var/www/html/Modules/Workflows/Entities/Workflow.php(563): Modules\\Workflows\\Entities\\Workflow->checkConditions()
#11 /var/www/html/Modules/Workflows/Entities/Workflow.php(529): Modules\\Workflows\\Entities\\Workflow->processForMailbox()
#12 /var/www/html/Modules/Workflows/Console/Process.php(41): Modules\\Workflows\\Entities\\Workflow::processWorkflowsForMailbox()
#13 [internal function]: Modules\\Workflows\\Console\\Process->handle()
#14 /var/www/html/overrides/laravel/framework/src/Illuminate/Container/BoundMethod.php(29): call_user_func_array()
#15 /var/www/html/overrides/laravel/framework/src/Illuminate/Container/BoundMethod.php(87): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#16 /var/www/html/overrides/laravel/framework/src/Illuminate/Container/BoundMethod.php(31): Illuminate\\Container\\BoundMethod::callBoundMethod()
#17 /var/www/html/overrides/laravel/framework/src/Illuminate/Container/Container.php(549): Illuminate\\Container\\BoundMethod::call()
#18 /var/www/html/vendor/laravel/framework/src/Illuminate/Console/Command.php(183): Illuminate\\Container\\Container->call()
#19 /var/www/html/vendor/symfony/console/Command/Command.php(255): Illuminate\\Console\\Command->execute()
#20 /var/www/html/vendor/laravel/framework/src/Illuminate/Console/Command.php(170): Symfony\\Component\\Console\\Command\\Command->run()
#21 /var/www/html/vendor/symfony/console/Application.php(992): Illuminate\\Console\\Command->run()
#22 /var/www/html/vendor/symfony/console/Application.php(255): Symfony\\Component\\Console\\Application->doRunCommand()
#23 /var/www/html/vendor/symfony/console/Application.php(148): Symfony\\Component\\Console\\Application->doRun()
#24 /var/www/html/vendor/laravel/framework/src/Illuminate/Console/Application.php(88): Symfony\\Component\\Console\\Application->run()
#25 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(121): Illuminate\\Console\\Application->run()
#26 /var/www/html/artisan(62): Illuminate\\Foundation\\Console\\Kernel->handle()
#27 {main}
"}
```

**Comment by MarcoWel** [comment]
See below:

![image](https://user-images.githubusercontent.com/80592577/190688763-23c96336-449b-457b-8580-e6f38fb31656.png)

**Comment by freescout-helpdesk** [comment]
Fixed in Custom Fields Module v1.0.7
